temporal-spring-boot-autoconfigure-alpha/src/main/java/io/temporal/spring/boot/autoconfigure/ClientAutoConfiguration.java

```diff
@@ -0,0 +1,90 @@
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.temporal.spring.boot.autoconfigure;
+
+import io.temporal.client.WorkflowClient;
+import io.temporal.client.WorkflowClientOptions;
+import io.temporal.serviceclient.WorkflowServiceStubs;
+import io.temporal.serviceclient.WorkflowServiceStubsOptions;
+import io.temporal.spring.boot.autoconfigure.properties.ClientProperties;
+import io.temporal.spring.boot.autoconfigure.properties.TemporalProperties;
+import io.temporal.testing.TestWorkflowEnvironment;
+import javax.annotation.Nullable;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.beans.factory.support.BeanDefinitionValidationException;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+/** Provides a client based on `spring.temporal.client` section */
+@Configuration
+@EnableConfigurationProperties(TemporalProperties.class)
+@ConditionalOnClass(name = "io.temporal.client.WorkflowClient")
+@ConditionalOnProperty(prefix = "spring.temporal", name = "client.target")
+public class ClientAutoConfiguration {
+
+  @Bean(name = "temporalWorkflowClient")
+  @ConditionalOnMissingBean(name = "temporalWorkflowClient")
+  public WorkflowClient client(
+      TemporalProperties temporalProperties,
+      @Qualifier("temporalTestWorkflowEnvironment") @Autowired(required = false) @Nullable
+          TestWorkflowEnvironment testWorkflowEnvironment) {
+    ClientProperties clientProperties = temporalProperties.getClient();
+    WorkflowServiceStubs workflowServiceStubs;
+    switch (temporalProperties.getClient().getTarget().toLowerCase()) {
+      case ClientProperties.TARGET_IN_PROCESS_TEST_SERVER:
+        if (testWorkflowEnvironment == null) {
+          throw new BeanDefinitionValidationException(
+              "Test workflow environment is not available. "
+                  + "Please make sure that you have enabled the 'temporal-test' module "
+                  + "and configured `spring.temporal.testServer.enabled: true`.");
+        }
+        return testWorkflowEnvironment.getWorkflowClient();
+      case ClientProperties.TARGET_LOCAL_SERVICE:
```

Comment:
There is none, but we have WorkflowServerStub.newLocalStubs, so I think it would be right to mirror it here.
The main argument for this is that it's much friendlier for new users to don't even worry about which port Temporal occupies by default.